scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -385,6 +385,14 @@
   com.netflix.fenzo.VirtualMachineLease
   (cpuCores [_] (or (offer-resource-scalar offer "cpus") 0.0))
   (diskMB [_] (or (offer-resource-scalar offer "disk") 0.0))
+  (getScalarValue [_ name] (or (double (offer-resource-scalar offer name)) 0.0))
+  (getScalarValues [_]
+    (reduce (fn [result resource]
+                 (if-let [value (:scalar resource)]
+                   (update-in result [(:name resource)] (fnil + 0.0 0.0) value)
```

Comment:
I wish--I observed cases in the wild, although I don't recall what triggered them. `value` can be `nil`. It may have had something to do with mesomatic doing funky stuff with resources, in that it always defines the `:scalar` key even when the `:type` is `:range` or something. I definitely needed that `fnil` to fix an NPE, though.